### DISCUSSION TITLE
How to update title of column

### DISCUSSION THREAD

**Discussion by GeorgeCodeHub** [nick-keller/react-datasheet-grid#172] [Q&A] [answered]
Hello there, I am trying to change the title of some columns when a State changes. However, when the state changes the titles don't seem to change. Is there a way to solve this problem?

Example of how I set it up:

```
const [selectedDate, setSelectedDate] = useState(new Date().toDateString());

const columns = useMemo(() => {
		const currentYear = formatDate(new Date(selectedDate), DatePatterns.YearOnly);

		return [
			{ ...keyColumn("Vehicles", intColumn), title: `Vehicles ${currentYear}` },
			{ ...keyColumn("Incidents", intColumn), title: `Incidents ${currentYear}` }
		];
	}, [selectedDate]);

<DataSheetGrid style={{ width: "100%" }} height={650} value={data} onChange={onDataSheetChange} columns={columns} lockRows={true} disableExpandSelection={true} />
```

Note: I am using version 4.1.0

**Comment by nick-keller** [comment, marked as answer]
Hello @GeorgeCodeHub,
Have you read this article: https://react-datasheet-grid.netlify.app/docs/performance/static-vs-dynamic
I would assume that you are using the static version, try switching to the dynamic version.
Let me know if that works ;)
